middleware/logger.go

```diff
@@ -13,22 +13,43 @@ import (
 	"time"
 )
 
+// DefaultLogFormatter contains all the data needed to construct a log message
+type DefaultLogFormatter struct {
+	buf bytes.Buffer
+}
+
+// LogFormatter allows you to customize this middleware's output format.
+type LogFormatter interface {
+	FormatRequest(r *http.Request) LogFormatter
+	FormatResponse(status, bytes int, elapsed time.Duration)
+	Log()
+	Recover(err error)
```

Comment:
Why do we need both FormatRequest and FormatResponse methods? Can't we just call one method only, on a way back from the underlying handlers? I don't see the reason why we'd need to store the intermediate log into a bytes.Buffer.

What's the reason for Log() and Recover()? Can't we live without them too?